src/gui/transferlistsortmodel.h

```diff
@@ -64,7 +66,9 @@ class TransferListSortModel final : public QSortFilterProxyModel
 
     TorrentFilter m_filter;
     mutable CachedSettingValue<int> m_subSortColumn;
+    mutable CachedSettingValue<bool> m_subSortColumnOrderAscending;
```

Comment:
That's a bit ugly name. I see most bool settings are either starts with `is` of with a verb. In this case I wanted to keep ` m_subSortColumn` prefix because both fields are basically one logical thing. And only then we have a verb (`Order`).
I hope this is acceptable.

Another thing that bugs me a bit is that Order is actually not a `bool` but an `enum` (with 2 values). Not sure if it is viable to store Qt::SortOrder in the settings (or if it is even possible). I decided it's better to keep it simple. It is highly unlikely that the enum will ever get its third value because the only ting possible is not to sort at all which is not relevant to sub-sorting

Reply:
>Another thing that bugs me a bit is that Order is actually not a `bool` but an `enum` (with 2 values).

It's generally a good idea. It makes code clearer.
>Not sure if it is viable to store `Qt::SortOrder` in the settings (or if it is even possible).

SettingsValue has overloaded methods to correctly handle enums that have associated "meta enum". 

>It is highly unlikely that the enum will ever get its third value 

The point of using enumerations instead of booleans is different.
E.g. here we really have two sorting order variants, ascending and descending, but not some property that can be either enabled/used or not.

Reply:
@glassez that is true and I agree. But what are you suggesting exactly? Is it to replace `CachedSettingValue<bool>` with `CachedSettingValue<Qt::SortOrder>` or to create some new enum akin to `BitTorrent::SessionSettingsEnums::MixedModeAlgorithm` and store it instead?

I first case settings are qt-dependent and it is possible that some future qt version will break compatibility (for example, encode acs/desc as `1`/`-1` instead of `0`/`1`). In second case it is just too much hassle to implement something that ["never been an explicitly declared feature"](https://github.com/qbittorrent/qBittorrent/issues/15002#issuecomment-849417585)

`bool` is both easy to support and future-qt-proof. Yes, it lacks some clarity of enums but considering it's limited use (basically one read and one write, limited to the same class) it shouldn't be an issue.

Reply:
>I first case settings are qt-dependent and it is possible that some future qt version will break compatibility (for example, encode acs/desc as 1/-1 instead of 0/1). 

It will be stored in human readable from: Ascending/Descending.

Reply:
@glassez Is that a roundabout way to say to use `CachedSettingValue<Qt::SortOrder>`? Sadly, it doesn't work as `Qt::SortOrder` underlying type is `unsigned int` and it fails static assert at `Utils::String::fromEnum()`: "Enumeration underlying type has to be int." What is the best way to address that?

Reply:
Sorry, I guess I was wrong... It's just a way of preserving some internal state. It is not intended for user editing. So there's no point in storing it user friendly. 
But I would still not use `bool`, and preferred `int`. This will shorten the names of the option/variable (`m_subSortOrder`). You can store a numeric representation of `Qt::SortOrder`. If you are afraid of depending on Qt, then you can use indirect values (although I doubt that there is any danger in this case, even if Qt changes `Qt::SortOrder` further, which is unlikely).

Reply:
@glassez this is now implemented.